src/drivers/imu/analog_devices/adis16507/ADIS16507.cpp

```diff
@@ -66,20 +67,22 @@ int ADIS16507::init()
 	int ret = SPI::init();
 
 	if (ret != PX4_OK) {
-		DEVICE_DEBUG("SPI::init failed (%i)", ret);
+		PX4_ERR("SPI::init failed (%i)", ret);
 		return ret;
 	}
 
-	return Reset() ? 0 : -1;
+	_state = STATE::RESET;
+	ScheduleDelayed(1_ms);
```

Comment:
Why?

Reply:
If you try to send the software reset command immediately after writing some other data the command won't get received. There needs to be slight delays between writes to give the ASIC time to process. I'll clean up the delays and comment them.